### DISCUSSION TITLE
EPS not being cropped properly

### DISCUSSION THREAD

**Discussion by AdamElkins** [ImageMagick/ImageMagick#4167] [Digital Image Processing]
I am having an issue converting EPS graphics. We have two subsets of graphics, one created in coreldraw, the other created in adobe illustrator, and the ones created in CorelDraw are problematic. When I call imagemagick via the command line and convert to png two graphics are created. The first one is a large graphic, with the actual graphic being tiny in the lower left corner. The second graphic looks correct. I have found that I can pass an index "imagemagick example.eps[1] example.png" and I get the 1 correct graphic. When I call identity, I get the following, which indicates there are two graphics in the eps, the first one is the large, incorrect one, and the second one is the correct one. In the specific use case I can't use the index, I also can't use the insert or delete parameters.

My question is this, are there other, better options to get the conversion to create a single graphic with the correct layer, that appears to ignore what I think is the bounding box?

 Format: EPT (Encapsulated PostScript with TIFF preview)
  Class: DirectClass
  Geometry: 612x792+0+0
  Resolution: 72x72
  Print size: 8.5x11
  Units: Undefined
  Colorspace: CMYK
  Type: ColorSeparation
  Endianness: Undefined
  Depth: 16/8-bit
  Channel depth:
    Cyan: 8-bit
    Magenta: 8-bit
    Yellow: 8-bit
    Black: 8-bit
  Channel statistics:
    Pixels: 484704
    Cyan:
      min: 0  (0)
      max: 47288 (0.721569)
      mean: 11.1108 (0.000169539)
      median: 0 (0)
      standard deviation: 592.432 (0.00903993)
      kurtosis: 3705.84
      skewness: 59.1297
      entropy: 0.00157625
    Magenta:
      min: 0  (0)
      max: 59624 (0.909804)
      mean: 12.1171 (0.000184896)
      median: 0 (0)
      standard deviation: 658.993 (0.0100556)
      kurtosis: 4097.35
      skewness: 61.5888
      entropy: 0.00141384
    Yellow:
      min: 0  (0)
      max: 59624 (0.909804)
      mean: 12.0599 (0.000184022)
      median: 0 (0)
      standard deviation: 656.103 (0.0100115)
      kurtosis: 4120.37
      skewness: 61.7129
      entropy: 0.00140603
    Black:
      min: 0  (0)
      max: 57825 (0.882353)
      mean: 13.6023 (0.000207557)
      median: 0 (0)
      standard deviation: 725.533 (0.0110709)
      kurtosis: 3709.02
      skewness: 59.1497
      entropy: 0.00157625
  Image statistics:
    Overall:
      min: 0  (0)
      max: 59624 (0.909804)
      mean: 12.2225 (0.000186504)
      median: 0 (0)
      standard deviation: 658.265 (0.0100445)
      kurtosis: 3981.06
      skewness: 60.8341
      entropy: 0.0014931
  Total ink density: 294.902%
  Colors: 39
  Histogram:Deleted this to shorten thread
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 612x792+0+0
  Dispose: Undefined
  Iterations: 0
  Scene: 0 of 2
  Compression: Undefined
  Orientation: Undefined
  Properties:
    date:create: 2021-09-07T22:51:01+00:00
    date:modify: 2021-09-07T22:51:01+00:00
    ps:HiResBoundingBox: 612x792+0+0
    ps:Level: PS-Adobe-3.0 EPSF-3.0
    signature: 55f6fdb837c1522c9a0aadfbc5fbb8f009e2c6223d2b675b230da46c606f7850
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 1.8491MiB
  Number pixels: 484704
  Pixels per second: 27.4326MP
  User time: 0.016u
  Elapsed time: 0:01.017
  Version: ImageMagick 7.1.0-5 Q16 x64 2021-08-22 https://imagemagick.org
Image:
  Filename: 0005845310.eps
  Format: TIFF (Tagged Image File Format)
  Mime type: image/tiff
  Class: PseudoClass
  Geometry: 61x58+0+0
  Resolution: 150x150
  Print size: 0.406667x0.386667
  Units: PixelsPerInch
  Colorspace: sRGB
  Type: Palette
  Endianness: LSB
  Depth: 4/16-bit
  Channel depth:
    Red: 16-bit
    Green: 16-bit
    Blue: 16-bit
  Channel statistics:
    Pixels: 3538
    Red:
      min: 0  (0)
      max: 15 (0.996109)
      mean: 13.5251 (0.901674)
      median: 14.9416 (0.996109)
      standard deviation: 3.80286 (0.253524)
      kurtosis: 6.0041
      skewness: -2.66817
      entropy: 0.446954
    Green:
      min: 0  (0)
      max: 15 (0.996109)
      mean: 13.3792 (0.891945)
      median: 14.9416 (0.996109)
      standard deviation: 4.08236 (0.272157)
      kurtosis: 5.04071
      skewness: -2.52707
      entropy: 0.462885
    Blue:
      min: 0  (0)
      max: 15 (0.996109)
      mean: 13.3707 (0.891379)
      median: 14.9416 (0.996109)
      standard deviation: 4.10231 (0.273487)
      kurtosis: 4.98773
      skewness: -2.52008
      entropy: 0.463003
  Image statistics:
    Overall:
      min: 0  (0)
      max: 15 (0.996109)
      mean: 13.425 (0.894999)
      median: 14.9416 (0.996109)
      standard deviation: 3.99584 (0.26639)
      kurtosis: 5.33407
      skewness: -2.57248
      entropy: 0.457614
  Colors: 6
  Histogram: Deleted this part to reduce length.
  Colormap entries: 16
  Colormap:... deleted this part to keep it simple.
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Matte color: grey74
  Background color: white
  Border color: srgb(223,223,223)
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 61x58+0+0
  Dispose: Undefined
  Iterations: 0
  Scene: 1 of 2
  Compression: None
  Orientation: TopLeft
  Properties:
    comment: image description
    date:create: 2021-09-07T22:51:01+00:00
    date:modify: 2021-09-07T22:51:01+00:00
    exif:Tag 59932: 
    signature: 29c8a01a5bc30f24d09c0a14328dcd6d696ea7120bc7f4cb2d58151ea292e04d
    tiff:alpha: unspecified
    tiff:endian: lsb
    tiff:photometric: palette
    tiff:rows-per-strip: 58
  Artifacts:
    verbose: true
  Tainted: False
  Filesize: 6296B
  Number pixels: 3538
  Pixels per second: 2821P
  User time: 1.578u
  Elapsed time: 0:02.254
  Version: ImageMagick 7.1.0-5 Q16 x64 2021-08-22 https://imagemagick.org
identify: Unknown field with tag 59932 (0xea1c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/965.
identify: Unknown field with tag 59932 (0xea1c) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/965.

**Comment by snibgo** [comment]
See https://imagemagick.org/script/defines.php , the option "-define eps:use-cropbox=true", which you can use before reading the EPS file. Does that help?

> In the specific use case I can't use the index, I also can't use the insert or delete parameters.

Why not?

> My question is this, are there other, better options to get the conversion to create a single graphic with the correct layer, that appears to ignore what I think is the bounding box?

You can tell IM to read just one image using [0] or [1]. Or you can read two images then "-delete" just one of them, if there is some condition that IM can test, eg the smallest image.

**Reply by AdamElkins** [reply]
It does not, I saw another forum post about a similar issue, it didnt help there either. 
https://github.com/ImageMagick/ImageMagick/discussions/2632

I am doing bulk conversion and not every graphic is guarenteed to have this issue, and may not need the first layer deleted, or the second image selected.

**Comment by snibgo** [comment]
> I am doing bulk conversion and not every graphic is guarenteed to have this issue, and may not need the first layer deleted, or the second image selected.

How do you know which files have this issue, and which image is the correct one?

**Reply by AdamElkins** [reply]
The main differences I am seeing between the two files is that the bad file has a first image, or page that has geometry that is way to large. The bounding box is the same size. They are created in CorelDraw. They seem to have a print size of 8.5x11 which is way too large. The Identify command reveals two distinct image definitions. The first image or layer is the large one and it almost looks like a print preview.

They also have the following two errors.
identify: Unknown field with tag 59932 (0xea1c) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/965.
identify: Unknown field with tag 59932 (0xea1c) encountered. `TIFFReadCustomDirectory' @ warning/tiff.c/TIFFWarnings/965.

**Reply by fmw42** [reply]
That is just a warning that there is some custom TAG in the file that ImageMagick does not understand. It is just a warning and does not affect the output.
